models/userroom.js

```diff
@@ -1,31 +1,35 @@
 Chatter.UserRoom = ChatterUserRoom = Astro.Class({
-    name: "ChatterUserRoom",
-    collection: new Mongo.Collection("chatteruserroom"),
-
-    fields: {
-        userId: {
-            type: "string"
-        },
-
-        roomId: {
-            type: "string"
-        },
-
-        count: {
-            type: "number",
-            default: function() {
-              return 0;
-            }
-        }
-    },
+  name: "ChatterUserRoom",
+  collection: new Mongo.Collection("chatteruserroom"),
 
-    events: {
+  fields: {
+    userId: {
+      type: "string"
+    },
 
+    roomId: {
+      type: "string"
     },
 
-    methods: {
+    count: {
```

Comment:
the count attribute keeps track of the unread messages for every room and user. It initially was a separate model, but I thought it would be cleaner and easier to include in userRoom object in order to avoid further joins. This will probably change if I rethink the way I have structured user and room with userRooms.